xnu: use python3Minimal instead of python3

reckenrode opened the pull request:
###### Description of changes

xnu does not require a Python that supports configd, so use a minimal one without support. Doing this prevents an infinite recuresion when building configd with xpc, which is needed to build with a newer LLVM.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

reckenrode committed:
xnu: use python3Minimal instead of python3

xnu does not require a Python that supports configd, so use a minimal
one without support. Doing this prevents an infinite recuresion when
building configd with xpc, which is needed to build with a newer LLVM.

wegank commented:
The change being `pkgs/os-specific/darwin/apple-sdk/default.nix`, how does it affect `darwin.xnu`?

reckenrode commented:
Looks like I might have cherry-picked the wrong commit. Let me dig into this to see what happened and whether this one is even still necessary.

reckenrode commented:
I’m not sure what happened here. At one point while working on a configd fix, I ran into an infinite recursion. I thought I modified `default.nix` for xnu, but I obviously didn’t. I dropped this commit from my local branch, and it still doesn’t have an infinite recursion, so I guess that’s fine.

However, I do have an actual fix for xnu. It fails to build with newer clangs due needing `-lc++abi` on one of the programs it builds. I’m going to close this PR and push the correct one. Thanks for the catch!
